src/main/java/dataMgmt/MetadataManager.java

```diff
@@ -131,21 +140,40 @@ public void putProjectMetadata(String projectRoot, ProjectMetadata metadata) {
                 // Only add it if filepath is valid (non-null)
                 if (f.getFilePath() != null) {
                     String filePath = Paths.get(projectRoot, f.getFilePath()).toAbsolutePath().toString().replace('\\', '/');
-                    putFileMetadata(filePath, f);
+                    putFileMetadata(filePath, metadata.getProjectID(), f);
                 }
             }
         }
     }
 
-    public void putFileMetadata(String filePath, FileMetadata metadata) {
+    public void putFileMetadata(String filePath, long projectID, FileMetadata metadata) {
         fileMetadataMap.put(filePath, metadata);
         fileIDtoFilePath.put(metadata.getFileID(), filePath);
+        fileIDtoProjectID.put(metadata.getFileID(), projectID);
+        ProjectMetadata meta = projectMetadataMap.get(projectIDtoRootPath.get(projectID));
+        if (meta.getFiles() == null) {
+            meta.setFiles(new FileMetadata[] {metadata});
+        } else {
+            ArrayList<FileMetadata> files = new ArrayList<>(Arrays.asList(meta.getFiles()));
+            if (!files.contains(metadata)) {
+                files.add(metadata);
+            }
+            meta.setFiles(files.toArray(new FileMetadata[0]));
```

Comment:
The reason it is an array is because of the JSON parsing. If Jackson will write List objects as javascript arrays correctly, we can change this. I'll look into it.